ext/hash/blake3/fetch_upstream_blake3.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
```

Comment:
just a personal opinion, but pcre2 nor opcache's JIT code are cloned. I do not see a particular reason to make an exception.

Reply:
@devnexen this script makes it easy to see exactly how blake3 1.5.0 was cloned, and should make it easy to update to future versions (like 1.6.0) in the future - IMO it's worthwhile to keep this script. idk if it should be part of `.github/actions/verify-generated-files/action.yml` tho (there's a discussion on that topic here https://github.com/php/php-src/pull/13194#discussion_r1458860073 )

Reply:
I do not see the appeal nor the necessity. Couple of counter arguments from the top of my head :
- does not make build easier, CI included.
- if this blake3 release contains bugs, the upgrade can be done in the same way the internal pcre2 library does. There is no necessity to upgrade versions if it brings nothing to php or goes against php's needs otherwise.
- if this contains bugs we are able to fix quickly but would take a long time (or worse getting refused) upstream, in your case we would need to patch it on the fly or forking the repo.

Reply:
@divinity76 The point of `verify-generated-files` is to avoid sneaking malicious code into generated files that are collapsed during code reviews that people don't generally look at. This risk doesn't exist here. IMO, if the intent is to avoid people mistakenly editing these files, the same issue already exists for all other copied projects, and should be solved universally (if at all).

Reply:
@iluuu1994 
>The point of verify-generated-files is to avoid sneaking malicious code into generated files that are collapsed during code reviews that people don't generally look at. This risk doesn't exist here. 

Interesting, thanks for the explanation! 

Maybe I snuck some malicious code into one of the upstream_blake3/* files? If I did, this script would actually remove it, and then `verify-generated-files` would notice it!

@devnexen 

>I do not see the appeal 

My main appeal is that if I want to update the bundled blake3 in the future, i can just change the 1 line `git clone --branch '1.5.0` with `git clone --branch '1.5.1'` and the script should do the rest.

>does not make build easier, CI included.

fair

>if this blake3 release contains bugs, the upgrade can be done in the same way the internal pcre2 library does.

How does upstream pcre2 fixes get applied to the internal pcre2 library? i don't know

>There is no necessity to upgrade versions if it brings nothing to php or goes against php's needs otherwise.

There's good updates sometimes, for example the latest v1.5.0 fixed a arm big-endian compile issue, v1.4.1 increased ARM performance by ~10-30%, 1.3.3 fixed a GCC6.1 AVX512 debug corruption issue.

>if this contains bugs we are able to fix quickly but would take a long time (or worse getting refused) upstream, in your case we would need to patch it on the fly or forking the repo.

Yeah that's a good point. If we get to that point, this script is useless. Anecdotal evidence, the only time I made a PR to BLAKE3, it was accepted in 11 hours ( https://github.com/BLAKE3-team/BLAKE3/pull/131 ), but it was a trivial one.

Reply:
> There's good updates sometimes, for example the latest v1.5.0 fixed a arm big-endian compile issue, v1.4.1 increased ARM performance by ~10-30%, 1.3.3 fixed a GCC6.1 AVX512 debug corruption issue.

Those are valid reasons to upgrade but again upgrades can be handled in same manner at the aforementioned php components. Note that I m not against having blake3 feature, I see it as a "nice to have", just the way of the dependency being handled.